functors_task_1/student/include/impl.h

```diff
@@ -40,9 +51,21 @@ struct MixingPolicy
      * Whiskey + SevenUp = SevenPlusSeven;
      * Others + Others = Oops;
      */
-    static Cocktail mix(/**???*/)
+    static Cocktail mix(const AlcoholDrink AlDrink,const NonAlcoholDrink NonAlDrink)
     {
+        if(AlDrink==AlcoholDrink::Gin && NonAlDrink==NonAlcoholDrink::LimeJuice){
```

Comment:
That's a question of a taste, but consider:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es70-prefer-a-switch-statement-to-an-if-statement-when-there-is-a-choice